File: Data Structures & Algorithms/non-overlapping-intervals/submission-1.cpp
class Solution {
   public:
    int eraseOverlapIntervals(std::vector<std::vector<int>>& intervals) {
        int res = 0;
        sort(intervals.begin(), intervals.end(),
             [](const vector<int>& x, const vector<int>& y) { return x[1] < y[1]; });
        int last = intervals[0][1];

        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i][0] < last) {
                res++;
            } else {
                last = intervals[i][1];
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/minimum-interval-including-query/submission-1.py
class Solution:
    def minInterval(self, intervals: List[List[int]], queries: List[int]) -> List[int]:
        # define interval size helper function
        # use heap to find the minimum interval
        intervals.sort(key=lambda x: x[0])

        res = {}
        i = 0
        heap = []
        for query in sorted(queries):
            while i < len(intervals) and intervals[i][0] <= query:
                heapq.heappush(heap, [self.intervalSize(intervals[i]), intervals[i][1]])
                i += 1

            while heap and heap[0][1] < query:
                heapq.heappop(heap)

            res[query] = heap[0][0] if heap else -1

        return [res[q] for q in queries]

    def intervalSize(self, interval):
        return interval[1] - interval[0] + 1